components/LoanCard/LoanCard.tsx

```diff
@@ -16,55 +16,71 @@ type LoanCardProps = {
   display?: 'expanded' | 'compact';
 };
 
-export function LoanCard({
-  loan,
-  selectedAddress,
-  display = 'expanded',
-}: LoanCardProps) {
-  const title = `View loan #${loan.id}`;
-
-  const tokenSpec: TokenURIAndID = useMemo(
-    () => ({
-      tokenURI: loan.collateralTokenURI,
-      tokenID: ethers.BigNumber.from(loan.collateralTokenId),
-      forceImage: true,
-    }),
-    [loan.collateralTokenId, loan.collateralTokenURI],
-  );
-
-  const maybeMetadata = useTokenMetadata(tokenSpec);
-
-  const relationship =
-    selectedAddress === loan.borrower ? 'borrower' : 'lender';
-  const attributes = useMemo(
-    () =>
-      display === 'expanded' ? (
-        <ExpandedAttributes loan={loan} />
-      ) : (
-        <CompactAttributes loan={loan} />
-      ),
-    [display, loan],
-  );
+export const LoanCard = memo(
+  ({ loan, selectedAddress, display = 'expanded' }: LoanCardProps) => {
+    const title = `View loan #${loan.id}`;
 
-  if (maybeMetadata.isLoading) {
-    return (
-      <LoanCardLoading id={loan.id.toString()}>
-        {selectedAddress && <Relationship>{relationship}</Relationship>}
-        {attributes}
-      </LoanCardLoading>
+    const tokenSpec: TokenURIAndID = useMemo(
+      () => ({
+        tokenURI: loan.collateralTokenURI,
+        tokenID: ethers.BigNumber.from(loan.collateralTokenId),
+        forceImage: true,
+      }),
+      [loan.collateralTokenId, loan.collateralTokenURI],
     );
-  } else {
-    return (
-      <LoanCardLoaded
-        id={loan.id.toString()}
-        title={title}
-        metadata={maybeMetadata.metadata}>
-        {selectedAddress && <Relationship>{relationship}</Relationship>}
-        {attributes}
-      </LoanCardLoaded>
+
+    const maybeMetadata = useTokenMetadata(tokenSpec);
+
+    const relationship =
+      selectedAddress === loan.borrower ? 'borrower' : 'lender';
+    const attributes = useMemo(
+      () =>
+        display === 'expanded' ? (
+          <ExpandedAttributes loan={loan} />
+        ) : (
+          <CompactAttributes loan={loan} />
+        ),
+      [display, loan],
     );
-  }
-}
+
+    if (maybeMetadata.isLoading) {
+      return (
+        <LoanCardLoading id={loan.id.toString()}>
+          {selectedAddress && <Relationship>{relationship}</Relationship>}
+          {attributes}
+        </LoanCardLoading>
+      );
+    } else {
+      return (
+        <LoanCardLoaded
+          id={loan.id.toString()}
+          title={title}
+          metadata={maybeMetadata.metadata}>
+          {selectedAddress && <Relationship>{relationship}</Relationship>}
+          {attributes}
+        </LoanCardLoaded>
+      );
+    }
+  },
+  (prevProps, nextProps) => {
+    if (prevProps.display !== nextProps.display) {
+      return false;
+    }
+
+    if (prevProps.selectedAddress !== nextProps.selectedAddress) {
+      return false;
+    }
+
+    if (!prevProps.loan.id.eq(nextProps.loan.id)) {
```

Comment:
this is the main optimization, we'd get a different loan object even though the values within it were the same.